src/extension/src/ProcessHandler.py

```diff
@@ -151,12 +151,17 @@ def __check_process_state(self, process, seq_no):
         unused_err = ""
         for retry in range(0, Constants.MAX_PROCESS_STATUS_CHECK_RETRIES):
             time.sleep(retry)
-            output, unused_err = process.communicate()
```

Comment:
Since @rane-rajasi originally wrote this and had made fixes/improvements over time, she needs to review to double-check nothing is regressing based on past validation experience. We'd discussed this in the morning but it looks like she couldn't get to it as she had a flight to catch late afternoon.

She'd said the other bug around bufferedreader appeared to be related to this too. Please sync with her.

Reply:
Approving. There are a few changes to be done with this fix, those will be included in another PR